packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -89,8 +88,13 @@ class PinCodeBloc extends $PinCodeBloc {
 
   @override
   ConnectableStream<dynamic> _mapToAuthenticatedState() => Rx.merge([
-        _digitsCountState.switchMap((digitsCount) =>
-            _checkPin(_pinCode.value, digitsCount).asResultStream()),
+        digitsCount
+            .switchMap((_) => pinCodeService.getPinLength().asStream())
+            .where(
+                (storedPinLength) => storedPinLength == _pinCode.value.length)
+            .flatMap((digitsCount) => _checkPin(_pinCode.value)
```

Comment:
Why we use the flagMap instead of switch map?

Reply:
No reason, reverted to switch map